src/mame/drivers/gigatron.cpp

```diff
@@ -70,7 +77,27 @@ void gigatron_state::machine_reset()
 {
 }
 
+uint16_t gigatron_state::fetch_outx()
```

Comment:
Why have this stub definition when the CPU implements this in a way more likely to work? (If I were writing this, I would probably make the outputs a device callback.)

Reply:
Because when I used the function that the CPU had for it, I would get a

cannot call member function without object

Error

Reply:
I was doing it like 

uint16_t light = gigatron_cpu_device::read_outx() & 0xF;

What was I doing wrong?

Reply:
That's the syntax for calling a non-member function in C++. gigatron_cpu_device:: read_outx is and has to be a member function.

Reply:
What would be the correct way to write it?

Reply:
Any help please? I tried using m_maincpu->read_outx() and gcc always says
'class cpu_device' has no member named 'read_outtx'